### ISSUE TITLE
`type-annotation-spacing` & `arrow-spacing`: conflicting against each other

### ISSUE THREAD

**Issue by Azuriru** [eslint-stylistic/eslint-stylistic#1192]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [ ] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [ ] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

https://github.com/user-attachments/assets/954d875a-b4b8-4f5e-a68b-96858891f3a6

Basically, they both should just play nice with each other, this is a recent bug but I don't know on who's side.

### Reproduction

https://stackblitz.com/edit/github-8pqpapsc

### System Info

```shell
System:
    OS: Windows 11 10.0.26200
    CPU: (16) x64 AMD Ryzen 7 7800X3D 8-Core Processor           
    Memory: 16.78 GB / 31.12 GB
  Binaries:
    Node: 22.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.22 - C:\Users\Robyn\AppData\Roaming\npm\yarn.CMD
    npm: 10.9.2 - C:\Program Files\nodejs\npm.CMD
    pnpm: 10.11.0 - C:\Users\Robyn\AppData\Local\pnpm\pnpm.EXE
    bun: 1.3.10 - C:\Users\Robyn\.bun\bin\bun.EXE
  npmPackages:
    eslint: ^10.0.3 => 10.0.3
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
I think this is similar to #1187. Please refer to https://github.com/eslint-stylistic/eslint-stylistic/issues/1187#issuecomment-4160667382

> Sorry for the confusion. This was caused by my incorrect migration strategy. I suspect this is the same as #1079.
>
> For now, you can set `arrow` to `ignore`, or try version `6.0.0-beta.3` (which removes the deprecated `arrow` option). Thanks.
>
> Related workaround:
> - https://github.com/eslint-stylistic/eslint-stylistic/issues/1079#issuecomment-3674885898
> - https://github.com/eslint-stylistic/eslint-stylistic/issues/1079#issuecomment-3674888061

**Comment by Azuriru** [comment]
I will close this issue, since you guys already have a reference one.
